src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -207,12 +206,18 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
       clustersByProject = clusterList.groupBy(_.googleProject)
       visibleClusters <- clustersByProject.toList.flatTraverse[Future, Cluster] { case (googleProject, clusters) =>
         val clusterList = clusters.toList
-        authProvider.canSeeAllClustersInProject(userInfo.userEmail, googleProject) flatMap {
+        authProvider.canSeeAllClustersInProject(userInfo.userEmail, googleProject).recover { case NonFatal(e) =>
+          logger.warn(s"Sam returned an exception for resource ${googleProject.value}. Filtering out from list results.", e)
+          false
```

Comment:
"list clusters" should not throw exceptions for individual clusters: it should filter them out.